File: AspectJDemo/app/src/main/java/com/example/aspectjdemo/MainActivity.java
package com.example.aspectjdemo;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.aspectjdemo.track.TrackEvent;
import com.example.aspectjdemo.track.TrackParameter;

/**
 * Created by cnting on 2022/7/29
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //这里还不能用语法糖
        findViewById(R.id.btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d("===>", "点击了");
            }
        });
//        testAfterReturning();
//        testAfterThrowing();
//        testAround();
//        testWithInCode();
//        test();
        trackMethod(100,"ct");
    }

    private TextView testAfterReturning() {
        return findViewById(R.id.tv);
    }

    private void testAfterThrowing() {
        TextView textView = null;
        textView.setText("aaa");
    }

    private void testAround() {
        TextView textView = null;
        textView.setText("aaa");
    }

    private void testWithInCode() {
        Person person = new Person();
        person.setAge(20);
    }

    private void test() {
        testAfterReturning();
        testAround();
        testWithInCode();
    }

    @TrackEvent(eventName = "点击按钮",eventId = "100")
    private void trackMethod(@TrackParameter("uid")int uid,String name){
        Log.d("===>","埋点");
    }
}

File: APTDemo/app/src/main/java/com/example/aptdemo/MainActivity.kt
package com.example.aptdemo

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.ioc_annotation.BindView

class MainActivity : AppCompatActivity() {

    @BindView(R.id.mainTextView)
    lateinit var textView: TextView

    @BindView(R.id.btn)
    lateinit var btn: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        com.example.ioc.ViewBinder.bind(this)
        Log.d("===>", "textView:$textView")
        Log.d("===>", "btn:$btn")
    }

}